src/features/userProfile/state/userProfileActions.ts

```diff
@@ -1,16 +1,37 @@
+import { ThunkDispatch } from 'redux-thunk'
+
+import { RootState } from "../../../store/store"
 import { ApiAction, ApiActionThunk, CALL_API } from "../../../store/api"
 
 import { NewSearchResult } from '../../../qri/search'
-import { UserProfileDatasetListParams } from '../../../qri/userProfile'
+import { UserProfile, UserProfileDatasetListParams } from '../../../qri/userProfile'
+import { mapVersionInfo } from "../../collection/state/collectionActions"
+import { USERPROFILE_SET } from "./userProfileState"
 
 export const searchPageSizeDefault = 25
 
-export function loadUserProfile (username: string): ApiActionThunk {
-  return async (dispatch, getState) => {
+export function loadUserProfile (username: string) {
+  return async (dispatch: ThunkDispatch<any, any, any>, getState: () => RootState) => {
+    const sessionUser = getState().session.user
+    if (sessionUser.username === username) {
+      return dispatch(setUserProfile(sessionUser))
+    }
```

Comment:
This is both to prevent us from double fetching, but also a work-around if you are working locally. If you are working locally, you may not have a cloud account, which means calling `fetchUserProfile` will come back with a 404 error.